packages/contracts-bedrock/scripts/outputs/DeleteOutput.s.sol

```diff
@@ -0,0 +1,208 @@
+     */
+    function setUp() external {
+        contracts[GOERLI] = ContractSet({
+            Safe: 0xBc1233d0C3e6B5d53Ab455cF65A6623F6dCd7e4f,
+            ProxyAdmin: 0x01d3670863c3F4b24D7b107900f0b75d4BbC6e0d,
+            L2OutputOracleProxy: 0xE6Dfba0953616Bacab0c9A8ecb3a9BBa77FC15c0
+        });
+    }
+
+    /**
+     * @notice Executes the gnosis safe transaction to delete an L2 Output Root.
+     */
+    function run(uint256 _index) external returns (bool) {
+        address _safe = contracts[GOERLI].Safe;
+        address _proxyAdmin = contracts[GOERLI].ProxyAdmin;
+        index = _index;
+        vm.startBroadcast();
+        bool success = _run(_safe, _proxyAdmin);
+        if (success) _postCheck();
+        return success;
+    }
+
+    /**
+     * @notice Follow up assertions to ensure that the script ran to completion.
+     */
+    function _postCheck() internal view override {
+        L2OutputOracle l2oo = L2OutputOracle(contracts[GOERLI].L2OutputOracleProxy);
+        Types.OutputProposal memory proposal = l2oo.getL2Output(index);
+        require(proposal.l2BlockNumber == 0, "DeleteOutput: Output deletion failed.");
+    }
+
+    /**
+     * @notice Test coverage of the script.
+     */
+    function test_script_succeeds() skipWhenNotForking external {
+        uint256 _index = getLatestIndex();
+        require(_index != 0, "DeleteOutput: No outputs to delete.");
+
+        index = _index;
+
+        address safe = contracts[GOERLI].Safe;
+        require(safe != address(0), "DeleteOutput: Invalid safe address.");
+
+        address proxyAdmin = contracts[GOERLI].ProxyAdmin;
+        require(proxyAdmin != address(0), "DeleteOutput: Invalid proxy admin address.");
+
+        address[] memory owners = IGnosisSafe(payable(safe)).getOwners();
+
+        for (uint256 i; i < owners.length; i++) {
+            address owner = owners[i];
+            vm.startBroadcast(owner);
+            bool success = _run(safe, proxyAdmin);
+            vm.stopBroadcast();
+
+            if (success) {
+                console.log("tx success");
+                break;
+            }
+        }
+
+        _postCheck();
+    }
+
+    function buildCalldata(address _proxyAdmin) internal view override returns (bytes memory) {
+        IMulticall3.Call3[] memory calls = new IMulticall3.Call3[](1);
+
+        calls[0] = IMulticall3.Call3({
+            target: oracle,
+            allowFailure: false,
+            callData: abi.encodeCall(
+                L2OutputOracle.deleteL2Outputs,
+                (index)
+            )
+        });
+
+        return abi.encodeCall(IMulticall3.aggregate3, (calls));
+    }
+
+    /**
+     * @notice Internal helper function to compute the safe transaction hash.
+     */
+    function _getTransactionHash(address _safe, address _proxyAdmin, address _l2oo, uint256 _index) internal returns (bytes32) {
+        // Ensure that the required contracts exist
```

Comment:
This code should be able to live in the safe builder? `buildCalldata` should read `_l2oo` and `_index` from storage

Reply:
Good call!